### DISCUSSION TITLE
Increase nixpkgs-update transparency

### DISCUSSION THREAD

**Discussion by rhendric** [nix-community/infra#791] [Ideas]
Hey all,

Periodically the nixpkgs-update repo gets complaints that r-ryantm hasn't been looking at an out of date package for a while. There are a number of reasons why this might be—personally, I'm very suspicious of the FIFO communication that the build02 config uses to bridge the fetchers and workers, as based on personal local testing I don't think that FIFO IO is multi-process-safe. But it could be something else, and the bigger problem is an overall lack of logging or transparency into what is happening on the machine.

I'd like to work on changing that by increasing the publicly-visible logging that nixpkgs-update does when it is instructed to look at a package. But I assume that you, the infra maintainers, have login access and a higher degree of visibility into output going to non-public logs, and I don't know if any changes I'd make would be redundant with information that you already have and just haven't made public. (And if so, could we instead redirect those logs to a public path?)

Does anyone have a good synthesis of what's going on here, or dos-and-don'ts for changing things?

**Comment by Mic92** [comment]
You can checkout https://r.ryantm.com/log/ for logs. Should be all public. If you have more suggestions you can checkout https://github.com/nix-community/infra/blob/bb45cc7190c157099e7824b91706dccbab8eed61/hosts/build02/nixpkgs-update.nix#L187 and make a pull request.

**Comment by rhendric** [comment]
Yeah I of course know about the public logs. But not everything nixpkgs-update outputs gets put in that path; some stuff is probably being dumped to stdout or stderr and ending up in the journal or syslog. I'd want to either redirect that output and also put it in the folder served as https://r.ryantm.com/log/, or find some other way to share it; but I don't want to disrupt whatever ops you guys use to monitor system health, which I don't know what that is.
